### DISCUSSION TITLE
Use more performant data grid to replace the raw html table

### DISCUSSION THREAD

**Discussion by mzy2240** [writer/writer-framework#8] [Ideas]
The idea is to keep some of the main components as performant as possible. A few candidates:

[Cheetah Grid](https://github.com/future-architect/cheetah-grid)
[DataGridXL2](https://github.com/DataGridXL/DataGridXL2)
[ToastUI Grid](https://github.com/nhn/tui.grid)

**Comment by ramedina86** [comment]
This is a great point, thanks for bringing it up. To be honest I hadn't tested big dataframes so I didn't experience any issues, but I did some tests and research and now I'm seeing it's a significant limitation.

Regarding next steps, I'll be working on a custom data grid as I also want to support sorting, interactions (e.g. select a row, trigger event) and edition. I want it to be light and highly coupled to Streamsync (events, styles, etc) so I'm not considering external libraries for this purpose. This is a priority and I expect to be part of 0.2.

In general, as "project philosophy" I'm refraining from using pre-built UI components. There's obvious downsides to this, but I think in the long run it'll pay off.

**Reply by mzy2240** [reply]
Yeah that make sense. You probably could take a look at the Cheetah Grid and start from somewhere there to have a minimal starting cost. It is licensed MIT so should be fine.

**Reply by ramedina86** [reply]
I developed a custom grid and added it to 0.1.7. It's quite basic now but should be easily expandable. I didn't use any code from Cheetah Grid but it was a good starting point to understand how grids work.

**Comment by matthiasautrata** [comment]
How about [tabulator](https://tabulator.info/)? This is inspired by learning [Holoviz Panel ](https://panel.holoviz.org/). They have lots of cool components that might serve as inspiration for your road map.
